com.unity.visualeffectgraph/Editor/Models/VFXGraph.cs

```diff
@@ -21,19 +21,32 @@ namespace UnityEditor.VFX
     [InitializeOnLoad]
     class VFXGraphPreprocessor : AssetPostprocessor
     {
-        void OnPreprocessAsset()
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
         {
-            bool isVFX = VisualEffectAssetModicationProcessor.HasVFXExtension(assetPath);
-            if (isVFX)
+            foreach (var assetPath in importedAssets)
             {
-                VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
-                if (resource == null)
-                    return;
-                VFXGraph graph = resource.graph as VFXGraph;
-                if (graph != null)
-                    graph.SanitizeForImport();
-                else
-                    Debug.LogError("VisualEffectGraphResource without graph");
+                bool isVFX = VisualEffectAssetModicationProcessor.HasVFXExtension(assetPath);
+                if (isVFX)
+                {
+                    VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
+                    if (resource == null)
+                        return;
+                    VFXGraph graph = resource.graph as VFXGraph;
+                    if (graph != null)
+                    {
+                        bool wasGraphSanitized = graph.sanitized;
+                        graph.SanitizeForImport();
+                        if (!wasGraphSanitized && graph.sanitized)
+                        {
+                            //Relaunch previously skipped OnCompileResource
+                            AssetDatabase.ImportAsset(assetPath, ImportAssetOptions.ForceUpdate);
```

Comment:
Collect & Process needed ImportAsset at the end of all Sanitize
See : https://unity.slack.com/archives/C47JPNNTZ/p1625218128286700?thread_ts=1623913383.239100&cid=C47JPNNTZ

Reply:
Resolved at https://github.com/Unity-Technologies/Graphics/pull/4971/commits/8d4cdecd684c64cf5e9bbb85b9f111abc7a481b3